libsoftwarecontainer/src/container.cpp

```diff
@@ -110,11 +110,16 @@ bool Container::initialize()
 {
     if (m_state < ContainerState::PREPARED) {
         std::string gatewayDir = gatewaysDir();
-        if (!createDirectory(gatewayDir)) {
+        if (!m_create.createDirectory(gatewayDir)) {
+            m_create.rollBack();
```

Comment:
This change makes little sense as any parent directories would have been cleaned anyway when the container dtor is called (and this is kinda fatal error).